src/characters/Character.ts

```diff
@@ -204,10 +204,13 @@ export default class Character extends Phaser.Physics.Arcade.Sprite {
   }
 
   isCharacterInArea(areas: Area[]) {
-    const area = areas.find((area) => area.contains(this.x, this.y))
-    if (area) {
-      area.handleCharacterInArea(this)
-    }
+    areas.forEach((area) => {
+      if (area.contains(this.x, this.y)) {
+        area.handleCharacterInArea(this)
+      } else {
+        area.handleCharacterNotInArea(this)
```

Comment:
I think this logic will break once you have more sound areas. Imagine, you have 2 sound areas, and that on the current moment the player is in the sound `area1`. `handleCharacterInArea` will be called and it will set `SoundSingleto.area` to `area1`. Then the foreach will run another cycle for the second area, lets called `area2`, the `handleCharacterNotInArea` will be called and it will clean `SoundSingleto.area`.

I suggest you use an array for `SoundSingleton.area`. You can add an area to the array once the players enters and remove it once the player leaves.